python/paddle/v2/fluid/evaluator.py

```diff
@@ -204,3 +204,37 @@ def eval(self, executor, eval_program=None):
             [precision], dtype='float32'), np.array(
                 [recall], dtype='float32'), np.array(
                     [f1_score], dtype='float32')
+
+
+class EditDistance(Evaluator):
+    """
+    Average edit distance error for multiple mini-batches.
+    """
+
+    def __init__(self, input, label, k=1, **kwargs):
+        super(EditDistance, self).__init__("edit_distance", **kwargs)
+        main_program = self.helper.main_program
+        if main_program.current_block().idx != 0:
+            raise ValueError("You can only invoke Evaluator in root block")
+
+        self.total_error = self.create_state(
+            dtype='float32', shape=[1], suffix='total')
+        self.batch_num = self.create_state(
+            dtype='float32', shape=[1], suffix='total')
+        error = layers.edit_distance(input=input, label=label)
+        error = layers.cast(x=error, dtype='float32')
+        mean_error = layers.mean(x=error)
```

Comment:
Better to make consistency with `Accuracy` evaluator, do not calculate the average mean error of current mini-batch.  Just accumulate the batch size and mini-batch error here.

Reply:
Fixed.